etc/sync_desi_spectro_dark.sh

```diff
@@ -36,30 +36,28 @@ for folder in dark_frames bias_frames; do
     echo `date --utc`
 done
 
-#- Get top-level CSV records
+#- Get top-level CSV records. Download CSV files to a temporary location.
+cd ${DESI_SPECTRO_DARK}/${VERSION}
+tmpdir=tmp-sync-`date "+%Y%m%d"`
+mkdir -p ${tmpdir}
+
 for csv in bias_table dark_table exp_daily_dark exp_dark_zero; do
     csvfile=${csv}.csv
-
-    #- Make a copy of the file in case of download problems.
-    if [ -s ${csvfile} ]; then
-        cp ${csvfile} ${csvfile}.bak
-    fi
     echo "Downloading ${csvfile} from ${URL}$"
 
-    cd ${DESI_SPECTRO_DARK}/${VERSION}
-
+    pushd ${tmpdir}
     wget --user=${user} --password=${passwd} --no-check-certificate ${URL}/${csvfile} -O ${csvfile}
-    if [ $? -ne 0 ]; then
-        #- When wget fails, restore the CSV file from its backup.
-        echo "wget FAILED for ${csvfile}."
-        if [ -s ${csvfile}.bak ]; then
-            echo "Restoring backup ${csvfile}."
-            mv ${csvfile}.bak ${csvfile}
-        fi
+    rc=$?
+    popd
+
+    if [ ${rc} -ne 0 ]; then
+        #- When wget fails, just emit a warning.
+        echo "wget FAILED to retrief ${csvfile}."
     else
-        #- Remove the backup CSV file under all other conditions.
-        rm -f ${csvfile}.bak
+        #- Copy the CSV file.
+        cp ${tmpdir}/${csvfile} .
```

Comment:
It looks like tmpdir is a subdirectory of the standard desi_spectro_dark directory, i.e. on the same file system.  If you do this as a "mv" instead of a "cp", then the rename will be essentially instantaneous instead of cp+rm which temporarily has a partially copied file in the final location.

Reply:
Update: maybe a "mv -f" to make it more robust to overwriting the previous file.